test/utils/realm.js

```diff
@@ -33,8 +34,10 @@ var kca = keycloakAdminClient(settings);
  */
 function createRealm (realmName) {
   var name = realmName || 'test-realm';
-  return kca.then((client) => {
-    return client.realms.create(parse(realmTemplate, name));
+  return kca.auth(settings).then(() => {
+    return kca.realms.create(parse(realmTemplate, name)).then(() => {
+      return kca.realms.findOne({ realm: realmName });
+    });
```

Comment:
I believe that sometimes the realm name can be sanitized by the API when created, so perhaps we can use the `realmName` that is returned from the endpoint?
```suggestion
    return kca.realms.create(parse(realmTemplate, name)).then(({ realmName: realm }) => {
      return kca.realms.findOne({ realm });
    });
```

That said, does the `create()` call not return the entire `RealmRepresentation`? If so we can simply omit the `findOne()` call.

Reply:
Since we're only using this for tests let's keep things as they are.